docs/user/customization.md

```diff
@@ -64,4 +64,167 @@ An example `worker.ign` is shown below. It has been modified to increase the HTT
 }
 ```
 
+## Install time customization for machine configuration
+
+**IMPORTANT**:
+
+- These customizations require using the `manifests` target that does not provide compatibility guarantees, for more information [check here](versioning.md#versioning).
+- This can affect upgradability of your cluster as the `machine-config-operator` can mark clusters tainted when user defined [MachineConfig][machine-config] objects are present in the cluster.
+
+In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. daemonset, deployment, etc). For example, if a user wants to run a logging agent on all hosts, in general the user should be deploying it as a daemonset object. However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster for example, due to a compliance mandate "the user must run auditing tools as soon as the operating system comes up". In this case, the admin user needs the ability to define a custom section of Ignition config (like the containing the systemd unit that runs the auditing container).
```

Comment:
nit: Potential rewords for this paragraph; take anything that sounds good:

> In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. DaemonSet, Deployment, etc).  For example, DaemonSets are the most stable way to run a logging agent on all hosts.  However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster.  For example, a compliance mandate like "the user must run auditing tools as soon as the operating system comes up" might require a custom systemd unit for an auditing container in the Ignition config for some or all nodes.

Reply:
changed to yours. I don't care enough ;)